cscs-checks/apps/cp2k/cp2k_check.py

```diff
@@ -13,11 +13,15 @@ def __init__(self, check_name, check_descr, **kwargs):
         self.executable = 'cp2k.psmp'
         self.executable_opts = ['H2O-256.inp']
 
+        energy = sn.extractsingle(r'\s+ENERGY\| Total FORCE_EVAL \( QS \) '
+                                  r'energy \(a.u.\):\s+(?P<energy>\S+)',
+                                  self.stdout, 'energy', float, item=-1)
         self.sanity_patterns = sn.all([
             sn.assert_found(r'PROGRAM STOPPED IN', self.stdout),
             sn.assert_eq(sn.count(sn.extractall(
                 r'(?P<step_count>STEP NUM)',
-                self.stdout, 'step_count')), 10)
+                self.stdout, 'step_count')), 10),
+            sn.assert_reference(energy, -4404.23228954148, -2.e-8, 1e-8)
```

Comment:
Does it make sense to use the `assert_bounded` with `abs` here? What's the meaning of the percentage of the reference energy?

Reply:
> Does it make sense to use the assert_bounded with abs here?
I think so too.

The code is compiled in double precision, so we should at least the double precision digits. I agree with @vkarak , I do not think we should have percentages for reference energies unless the code is compiled with a given precision and it prints more than it can handle. Which I think, it is not the case.